hack/vendor.sh

```diff
@@ -1,14 +1,41 @@
 #!/usr/bin/env bash
-
-# This file is just wrapper around 'go mod vendor' tool.
+#
+# This file is just a wrapper around the 'go mod vendor' tool.
 # For updating dependencies you should change `vendor.mod` file in root of the
 # project.
 
 set -e
-set -x
 
 SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 "${SCRIPTDIR}"/go-mod-prepare.sh
```

Comment:
Can we make this part of the `tidy` and `vendor` functions?

We should also have a `cleanup` that removes the `go.mod` afterwards; I keep running into the `go.mod` being left behind, and depending on wether it's there (or not) running tests within my IDE works or breaks.

I recall we had some discussion around that when we switched to `vendor.mod`, but having used it for some time, I think leaving the `go.mod` behind is causing more issues than it solves.

Reply:
I personally prefer to leave the uncommitted `go.mod` -- it lets me use VS Code or Goland without having to maintain (and sync) a GOPATH. I think this is quite important to support as we want to make sure that contributors are using the source code from `vendor/` and not from a `go get` that is ignorant of modules.

This does lead to a disconnect where `GO111MODULE=off` is used on the command line, and `GO111MODULE=on` is used in the IDE to discover vendor code, but I think that it is fine as the usage in the IDE is just to discover external modules.

We can add this script to the functions if you want to avoid running it on `help`, however, my thoughts are that we should actually double down on it (that is to say, the uncommitted `go.mod` should **always** be present) and having this as a per-requisite mandatory step is fine.

That is because I think the [IDE setup docs](https://github.com/moby/moby/blob/master/docs/contributing/set-up-ide.md#configure-project-for-ide-optional) are deeply flawed and should instead prefer to use a Go Modules-based approach (so, running `./hack/go-mod-prepare.sh` would be a vital step in setting up an IDE) as using a GOPATH means you may have different versions of dependencies from the vendoring.

Reply:
`github.com/moby/moby` is not a Go module. We want to make it one eventually, but we are not there yet. Leaving an uncommitted `go.mod` screws up IDEs which are set up properly to open the project in GOPATH mode. [The vendor tree, if present, is used during `go build` in both GOPATH mode and module mode.](https://golang.org/s/go15vendor) **`go.mod` is _not_ necessary to achieve reproducible builds using vendored dependencies.**

I second @thaJeztah; leaving an uncommitted `go.mod` causes more issues than it solves.

Reply:
Discussed offline with Cory; consensus was we're both right somewhat, but that Goland can be made to work as expected in GOPATH mode -- it's just hidden behind another unintuitive checkbox. I've started a follow-up PR (#44497) that builds on this one, and prevents uncommitted go.mod.